core/src/main/java/org/keycloak/representations/idm/authorization/UmaPolicyRepresentation.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2016 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.representations.idm.authorization;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * @author <a href="mailto:federico@martel-innovate.com">Federico M. Facca</a>
+ */
+
+public class UmaPolicyRepresentation extends AbstractPolicyRepresentation {
```

Comment:
in this way we would not "simplify" the payload as discussed in first instance. i understand that this may simplify lot of coding, but may be less friendly from the point of view of creating policies. as you prefer.

Reply:
I'm not sure either :) Let me think a bit more about that, maybe we can come up with something even easier. For instance, if we are supporting user, role, group and client policies, we could just use something like:

```json
{
    "scopes": ["read"],
    "users": ["a", "b"],
    "groups": ["c", "d"],
    "roles": ["e", "f"]
}
```

Reply:
while at creation time many things can be implied, the internal representation is still richer, and some how we have to deal with that (or with assumptions, or with "optional fields").

currently, for example to share with more groups, you will create something like that via the api:
```
{
    "scopes": [
        "read"
    ],
    "policies": [
	{
        "type": "group",
        "subjects": [
            "a759a3a2-95b9-4f88-9c5e-c2b9ea2f1211",
            "d14d66e5-3341-439a-9da4-60944487a158"
        ]
    }
    ]
}
```
or for users
```
{
    "scopes": [
        "read"
    ],
    "policies": [
	{
        "type": "user",
        "subjects": [
            "a759a3a2-95b9-4f88-9c5e-c2b9ea2f1211",
            "d14d66e5-3341-439a-9da4-60944487a158"
        ]
    }
    ]
}
```
The response will include the enriched json (including optional parameters - but probably we can have a way to not return them, unless they are different from the "default"):
```
{
    "logic": "POSITIVE",
    "decisionStrategy": "UNANIMOUS",
    "scopes": [
        "read"
    ],
    "policies": [
        {
            "id": "94c61cd8-6d64-4e08-91e7-7066809db190",
            "type": "group",
            "logic": "POSITIVE",
            "decisionStrategy": "UNANIMOUS",
            "subjects": [
               "a759a3a2-95b9-4f88-9c5e-c2b9ea2f1211",
               "d14d66e5-3341-439a-9da4-60944487a158"
            ],
            "options": {
                "groupsClaim": [
                    "groups"
                ]
            }
        }
    ],
    "id": "d59d7e08-1448-4a65-894a-ee921987b348"
}
```

Reply:
Yeah, I'll agree with you. Let's keep this structure.

+1 to not return "options" if they are basically the defaults. And I think does not make sense to return "id" and "decisionStrategy". The "id" because if you want to update the "group" policy above we could just update the existing "group" policy, right ?